### DISCUSSION TITLE
Workflow suggestions for a note-taking newbie

### DISCUSSION THREAD

**Discussion by c-c-k** [jakewvincent/mkdnflow.nvim#190] [General]
Hi,

I'm trying to start/restart using a notebook/personal knowledge base/zettelkasten/second brain/etc..

I've looked through the many related vim/nvim plugins, got a bit lost and overwhelmed, started to write something of my own, got even more lost and overwhelmed, looked at some of the related plugins again and decided that mkdnflow is probably the closest to what I want.

However there are some basic note-taking workflow issues that I can't get my head around and was hoping more experienced note-takers could give me their insights on.

### Reference Links

Due to vim/nvim's long standing issues with text concealment, reference style links seem like the best option for vim/nvim as they don't induce much visual clatter when left unconcealed, however as far as I've noticed, none of the existing plugins offer the possibility of using them as the preferred add-link function/command.

So I was wondering:

1. Is the visual clatter introduced by normal (`[...](...)`) or wiki (`[[...|...]]`) links isn't actually that bad after using them for a while?
2. Or is there some serious downside to dumping all the reference link destinations (`[...]: ...`) at the end of the markdown file?
3. Would it be hard to make mkdnflow add reference links?

### Viewing/editing notes on a mobile & viewing notes with images and the like

Images, math formulas, graphs, charts, etc.. are things that as far as I know vim/nvim can't show/render inline.

While working on a note in vim/nvim using one of the markdown preview plugins seems like a good answer.

But what would be a good solution for occasionally viewing/editing the notes on a mobile phone.

### Link path destinations

When following a link with an absolute path (`/topic/sub-topic/note.md`), Github and vimwiki treat the root dir (`/`) as the root dir of the notebook/wiki/repository.

In my own attempted plugin I've made it so that when creating a link with a relative input path (`../another-sub-topic/other-note.md`) the input link would be interpreted relative to the current note, but the actual link destination inserted in the note would be the absolute path within the notebook (`/topic/another-sub-topic/other-note.md`).

This seemed a good idea to me because it would allow to clearly see the link destinations later on and would make refactoring the notebook easier.

1. Is there some serious downside to that sort of an approach?
2. Can this behavior be achieved with mkdnflow?

### Inter-notebook links

Vimwiki allows to create links from one wiki notebook to another with a `vw-anotherwiki:/path/to/note` URI. 

1. Is there some serious downside to that sort of an approach?
2. Can this behavior be achieved with mkdnflow?

Anyway, sorry for the long post, thanks in advance for any insights,
c-c-k.

**Comment by life00** [comment]
Hi @c-c-k. I'll try to answer some of your questions based on my experience and workflow.

You can enable concealment of links using mkdnflow.nvim in the config.

Depends on the platform. If you are using android then I may recommend [Obsidian](https://obsidian.md/) (proprietary) or [Markor](https://github.com/gsantner/markor) (open source). You can sync everything across your devices using syncthing. If you are using iOS then I express my sincere condolences; iOS ecosystem is pretty bad in terms of flexibility and idk any apps. There is Joplin but it uses its own special format for storing notes. Actually you might try using Obsidian for iOS. 

Regarding navigation I don't really use this type of *web of links* across notes. I just use my file manager. But relative paths is the optimal thing if you want your notes to be portable across systems.

sorry, idk

I hope this was helpful. Feel free to ask follow up questions.

**Reply by c-c-k** [reply]
Hi @life00, thank you for the reply.

> You can enable concealment of links using mkdnflow.nvim in the config.

My problem with concealment is that vim&nvim handle it badly (i.e. concealed text that gets soft-wrapped creates multi-line empty space, and, navigation across concealed text sometimes works unintuitively). I don't remember if I tried enabling the concealment of links feature in the mkdnflow config, does it overcome the above problems? (sorry for asking instead of just checking, my nvim setup is a mess right now)

> Regarding navigation I don't really use this type of web of links across notes. I just use my file manager. But relative paths is the optimal thing if you want your notes to be portable across systems.

I dunno, the Zettel idea does sound nice but then again it might not really give that much of an advantage compared to just opening another related note with the file manager if it's "nearby" and searching for it if it's not. In this case inter-note links become fundamentally redundant, also, since you might use your notes on more than one device, you can't reliably link to anything outside of your notes, so local links are pointless as a whole and it doesn't really matter if they are relative or absolute, no?

> Depends on the platform. If you are using android then I may recommend [Obsidian](https://obsidian.md/) (proprietary) or [Markor](https://github.com/gsantner/markor) (open source). You can sync everything across your devices using syncthing. If you are using iOS then I express my sincere condolences; iOS ecosystem is pretty bad in terms of flexibility and idk any apps. There is Joplin but it uses its own special format for storing notes. Actually you might try using Obsidian for iOS.

I've tried Obsidian, but working on the basic assumption "I want to use nvim as my note editor & manager", using Obsidian and it's fairly large ecosystem of plugins feels like it would just complicate matters as it would require first learning Obsidian and it's plugins and then figuring how do I divide the note-taking workflow between obsidian and nvim.


Anyway, since I'm not really sure what I want I decided to try and build a plugin of my own, not because I think I can do something that this excellent plugin doesn't, but because I'm hoping that the process of building a plugin in and of itself will help me to sort out my head.

Thanks again and sorry for taking your time.

**Reply by life00** [reply]
> My problem with concealment is that vim&nvim handle it badly (i.e. concealed text that gets soft-wrapped creates multi-line empty space, and, navigation across concealed text sometimes works unintuitively). I don't remember if I tried enabling the concealment of links feature in the mkdnflow config, does it overcome the above problems? (sorry for asking instead of just checking, my nvim setup is a mess right now)

Yeah its pretty good. mkdnflow provides *some* concealment (including link), but you may also take a look at other plugins like vim-markdown and vim-pandoc-syntax. Also if you are willing it is possible to configure concealment using treesitter, but it would require some time https://www.youtube.com/watch?v=8fCkt5qgHw8 (there is many other videos).

> I dunno, the Zettel idea does sound nice but then again it might not really give that much of an advantage compared to just opening another related note with the file manager if it's "nearby" and searching for it if it's not. In this case inter-note links become fundamentally redundant, also, since you might use your notes on more than one device, you can't reliably link to anything outside of your notes, so local links are pointless as a whole and it doesn't really matter if they are relative or absolute, no?

Yes, unfortunately this is one of the downsides of my setup. I don't have time to care about correctly linking everything using relative links. I also only link supplementary materials to notes (e.g. PDFs), and usually never notes. Again I categorize everything in directories. I have a huge notes git repository where I store my notes in different directories. If necessary I provide additional documentation what this directory is about inside that directory. I usually just ignore the problem of broken paths because my main device is my laptop and being able to access my notes from my phone is just a bonus convenience in case I urgently need something to check. But I don't actually use it like a note editor.

> I've tried Obsidian, but working on the basic assumption "I want to use nvim as my note editor & manager", using Obsidian and it's fairly large ecosystem of plugins feels like it would just complicate matters as it would require first learning Obsidian and it's plugins and then figuring how do I divide the note-taking workflow between obsidian and nvim.

I recommended Obsidian because it is the most polished app and it probably supports your note structure (just a folder with subfolders and notes). You don't need to customize it. You can use syncthing with it (sync notes to local android dir and set the "vault" as that directory in obsidian).

> Anyway, since I'm not really sure what I want I decided to try and build a plugin of my own, not because I think I can do something that this excellent plugin doesn't, but because I'm hoping that the process of building a plugin in and of itself will help me to sort out my head.

Good luck.

**Reply by life00** [reply]
@c-c-k Just keep in mind that there might be a simpler and more elegant way of doing things. But of course it is subjective. For example recently I simplified my citation setup. I previously used papis with its database and also some neovim plugin for it, but then I realized that its too much unnecessary work and abstractions. I just don't put the references into the database. Instead a more ergonomic workflow for me was to put all the references and papers inside the project itself. References inside `.bib` also with stuff like https://github.com/aspeddro/cmp-pandoc.nvim completion, and papers just in a subdirectory.

**Reply by c-c-k** [reply]
Hi again @life00, thank you for your insights and your time, I do indeed have a crippling tendency to over-complicate things, I will consider your insights and perhaps they will light my path, for the moment I'm on the edge of surrendering to ChatGPT like most of the people around me who are much more successful than me did...
